pkg/network/dns/cache.go

```diff
@@ -143,35 +139,15 @@ func (c *reverseDNSCache) Get(ips []util.Address) map[util.Address][]Hostname {
 	}
 
 	// Update stats for telemetry
-	c.lookups.Add(int64(len(resolved) + len(unresolved)))
-	c.resolved.Add(int64(len(resolved)))
-	c.oversized.Add(int64(len(oversized)))
+	lookups.Add(float64((len(resolved) + len(unresolved))))
+	resolved_tel.Add(float64(len(resolved)))
+	oversized_tel.Add(float64(len(oversize)))
 
 	return resolved
 }
 
 func (c *reverseDNSCache) Len() int {
-	return int(c.length.Load())
-}
-
-func (c *reverseDNSCache) Stats() map[string]int64 {
-	var (
-		lookups   = c.lookups.Load()
-		resolved  = c.resolved.Load()
-		added     = c.added.Load()
-		expired   = c.expired.Load()
-		oversized = c.oversized.Load()
-		ips       = int64(c.Len())
-	)
-
-	return map[string]int64{
-		"lookups":   lookups,
-		"resolved":  resolved,
-		"added":     added,
-		"expired":   expired,
-		"oversized": oversized,
-		"ips":       ips,
-	}
+	return int(length.Load())
```

Comment:
This is not `len(c.data)`? You can remove the wrapper usage if that is the case.

Reply:
yep you are right. I'll change this to a gauge